view/frontend/templates/button_product_page.phtml

```diff
@@ -333,18 +340,66 @@ foreach ($block->getAdditionalCheckoutButtonAttributes() as $attrName => $attrVa
             };
 
             var cart;
-            var setupProductPage = function () {
+            var setupProductPage = function (additionalData) {
                 cart = {
                     currency: "<?= /* @noEscape */ $block->getStoreCurrencyCode(); ?>",
-                    items: getItemsData()
+                    items: getItemsData(additionalData)
                 };
-
                 // if connect.js is not loaded postpone until it is
                 whenDefined(window, 'BoltCheckout', function(){
                     BoltCheckout.configureProductCheckout(cart, hints, callbacks);
                 }, 'configureProductCheckout');
             };
 
+            // Extend_Warranty 3-th party module support
+            // collecting selected warranty plans and add as additional product options in request
+            var extendWarrantyCollectPlans = function () {
+                // check if Extend_Warranty global variable = "Extend" is defined
+                if (window.Extend !== undefined && window.Extend.buttons !== undefined) {
```

Comment:
1) Let's do an early exit instead
2) What do you think about checking in PHP if the module is enabled and "print" empty function in this case?